crates/ibc/src/core/ics02_client/events.rs

```diff
@@ -21,262 +21,324 @@ pub const CLIENT_TYPE_ATTRIBUTE_KEY: &str = "client_type";
 /// The content of the `key` field for the attribute containing the height.
 pub const CONSENSUS_HEIGHT_ATTRIBUTE_KEY: &str = "consensus_height";
 
+pub const CONSENSUS_HEIGHTS_ATTRIBUTE_KEY: &str = "consensus_heights";
+
 /// The content of the `key` field for the header in update client event.
 pub const HEADER_ATTRIBUTE_KEY: &str = "header";
 
-/// NewBlock event signals the committing & execution of a new block.
-// TODO - find a better place for NewBlock
-#[derive(Debug, Serialize, Clone, Copy, PartialEq, Eq)]
-pub struct NewBlock {
-    pub height: Height,
+#[derive(Debug)]
+struct ClientIdAttribute {
+    client_id: ClientId,
 }
 
-impl NewBlock {
-    pub fn new(h: Height) -> NewBlock {
-        NewBlock { height: h }
+impl From<ClientIdAttribute> for Tag {
+    fn from(attr: ClientIdAttribute) -> Self {
+        Tag {
+            key: CLIENT_ID_ATTRIBUTE_KEY.parse().unwrap(),
+            value: attr.client_id.to_string().parse().unwrap(),
+        }
     }
-    pub fn set_height(&mut self, height: Height) {
-        self.height = height;
+}
+
+impl From<ClientId> for ClientIdAttribute {
```

Comment:
I would suggest using `derive_more::From` to derive these `From` implementations for all attributes.

Reply:
Good idea! [bc9274a](https://github.com/cosmos/ibc-rs/pull/138/commits/bc9274a6fa2f46ecdda60abe4bfa412603c9423b)